src/components/TopProject.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState } from "react";
+import SocialMedia from "components/SocialMedia/SocialMedia";
+import { SocialMediaColor } from "../utility/SharedStyles";
+import { faGithub } from "@fortawesome/free-brands-svg-icons";
+import * as S from "./TopProject.styles";
+
+interface topProjectProps {
+    image?: string;
+    github: string;
+    description: string;
+    name: string;
+    year: string;
+}
+function TopProject(props: topProjectProps) {
+    const [descriptionIsVisible, setDescriptionIsVisible] = useState(false);
+    const [nameIsVisible, setNameIsVisible] = useState(false);
+
+    function mouseLeave() {
+        setDescriptionIsVisible(false);
+        setNameIsVisible(false);
+    }
+    function mouseEnter() {
+        setDescriptionIsVisible(true);
+        setNameIsVisible(true);
+    }
```

Comment:
What are these states and mouse callbacks meant to achieve? I only see the same effects as just using the data-aos animations in the preview.